File: data/src/main/java/com/smedialink/tokenplussteamid/data/dao/BaseDao.java
package com.smedialink.tokenplussteamid.data.dao;

import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;

import java.util.List;

/**
 * Created by six_hundreds on 31.01.18.
 */

public interface BaseDao<T> {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(T entity);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(List<T> entity);
}

File: data/src/main/java/com/smedialink/tokenplussteamid/data/dao/BaseDao.kt
package com.smedialink.tokenplussteamid.data.dao

import android.arch.persistence.room.Insert

/**
 * Created by six_hundreds on 24.01.18.
 */
interface BaseDao<T> {

    @Insert
    fun insert(entity: T)

    @Insert
    fun insert(entities: ArrayList<T>)

//    @Delete
//    fun deleteById(id: Number)

//    @Update
//    fun update(entity: T)
}